addons/project/models/project.py

```diff
@@ -699,6 +700,28 @@ def unlink(self):
         analytic_accounts_to_delete.unlink()
         return result
 
+    @api.model
+    def _search(self, args, offset=0, limit=None, order=None, count=False, access_rights_uid=None):
+        order_items = [order_item.strip().lower() for order_item in (order or self._order).split(',')]
+        new_order, item_index, desc = [], -1, False
+        for index, item in enumerate(order_items):
+            if 'is_favorite' in item:
+                item_index = index
+                if 'desc' in item:
+                    desc = True
+            else:
+                new_order.append(item)
+        new_order = ', '.join(new_order)
+        query = super()._search(args, offset, limit, new_order, count, access_rights_uid)
+        if item_index != -1:
+            query.order = ', '.join([f"""
+                id IN (
+                    SELECT project_id
+                    FROM project_favorite_user_rel
+                    WHERE user_id = {self.env.uid}
+                )""" + " DESC" * desc, new_order])
```

Comment:
The uid can be passed in the params of the query. 🙂 
Also, we have to add the is_favorite order in the right place since we have the index in which it was there in the order passed in parameter. 🙂 
Finally, we cannot really use the new_order parameter to set the right order; we should use the query.order instead since the order is built in the `generate_order_by` method. 🙂
```suggestion
            query_order_list = query.order.split(',')
            is_favorite_order = f"""
                id IN (
                    SELECT project_id
                    FROM project_favorite_user_rel
                    WHERE user_id = %s
                ) {"DESC" if desc else "ASC"}"""
            query_order_list.insert(item_index, is_favorite_order)
            query.order = ', '.join(query_order_list)
            query._where_params.append(self.env.uid)
```